pulp_ansible/app/galaxy/v3/viewsets.py

```diff
@@ -0,0 +1,32 @@
+from rest_framework.pagination import PageNumberPagination
+from rest_framework import viewsets
+
+from pulp_ansible.app.galaxy.v3.serializers import (
+    CollectionVersionSearchListSerializer,
+)
+
+from pulp_ansible.app.models import (
+    CrossRepositoryCollectionVersionIndexView,
+)
+
+from pulp_ansible.app.galaxy.v3.filters import CollectionVersionSearchFilter
+
+
+class CollectionVersionSearchViewSetPagination(PageNumberPagination):
+    page_size = 100
+    page_size_query_param = "page_size"
+    max_page_size = 1000
+
+
+class CollectionVersionSearchViewSet(viewsets.ModelViewSet):
+    """
+    A viewset for cross-repo searches.
+    """
+
+    serializer_class = CollectionVersionSearchListSerializer
+    pagination_class = CollectionVersionSearchViewSetPagination
```

Comment:
This needs to use the same pagination class as the rest of the api endpoints (offset/limit)